### PR TITLE
bpo-24813: IDLE: Add icon to help_about

### PR DESCRIPTION
**Pull request by csabella**
Based on part of a patch originally by Mark Roseman.

### CODE REVIEW

File: Lib/idlelib/help_about.py [hunk 1]
```diff
@@ -63,6 +63,15 @@ def create_widgets(self):
         header = Label(frame_background, text='IDLE', fg=self.fg,
                        bg=self.bg, font=('courier', 24, 'bold'))
         header.grid(row=0, column=0, sticky=W, padx=10, pady=10)
+
+        tk_patchlevel = self.tk.call('info', 'patchlevel')
+        ext = '.png' if tk_patchlevel >= '8.6' else '.gif'
+        iconfn = os.path.join(os.path.abspath(os.path.dirname(__file__)),
```
Comment (terryjreedy) [inline Lib/idlelib/help_about.py]: I suspect 'iconfn' abbreviates 'icon file name'.  Too obscure; I'd prefer 'icon' here and below.

File: Lib/idlelib/idle_test/test_help_about.py [hunk 2]
```diff
@@ -33,6 +34,12 @@ def test_dialog_title(self):
         """Test about dialog title"""
         self.assertEqual(self.dialog.title(), 'About IDLE')
 
+    def test_dialog_logo(self):
+        """Test about dialog logo."""
+        path, file = os.path.split(self.dialog.icon_image['file'])
+        fn, ext = os.path.splitext(file)
+        self.assertEqual(fn, 'idle_48')
```
Comment (terryjreedy) [inline Lib/idlelib/idle_test/test_help_about.py:41]: This is the best I can think of too.  The real test is running help_about for the htest.  I did that to experiment with different backgrounds.

**Review by terryjreedy — CHANGES_REQUESTED** [review]
I forgot to specify icon to left (this is standard), 'IDLE' to its right (in current center).  With this position switch and 1 name change, PR will be ready to merge.

File: Lib/idlelib/idle_test/test_help_about.py (hunk 2 above)
Comment (csabella) [reply Lib/idlelib/idle_test/test_help_about.py:41]: Thanks.  I wasn't sure if I should mock the PhotoImage just to show that the widget is created.

**Comment by csabella** [comment]
Thanks.  Looks better now.  :-)

**Review by terryjreedy — APPROVED** [review]

**Pull request merged**